tests/lib/modem_trace/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  unity.modem_trace:
+    platform_allow: qemu_cortex_m3 native_posix
```

Comment:
Please add here also entry for `integration_platforms` (it allows to perform this test in CI every time when somebody will create PR):
```
    integration_platforms:
      - qemu_cortex_m3
      - native_posix
```

Reply:
Is that applicable for unit tests too?

Reply:
Yes, I think that there is no problem for unit tests. After add this `integration_platforms` entry to this yaml file I am able to  run those tests by this command (from `ncs/zephyr` directory):
```
scripts/twister --integration -T ../nrf/tests/lib/modem_trace/
```
And I observed that tests pass both on `qemu_cortex_m3` and `native_posix` platforms:

> *** Booting Zephyr OS build v2.7.0-ncs1-4-gf295fb7c67ca  ***
> src/modem_trace_test.c:200:test_modem_trace_start_when_not_initialized:PASS
> src/modem_trace_test.c:205:test_modem_trace_init_uart_transport_medium:PASS
> src/modem_trace_test.c:215:test_modem_trace_init_rtt_transport_medium:PASS
> src/modem_trace_test.c:227:test_modem_trace_start_coredump_only:PASS
> src/modem_trace_test.c:233:test_modem_trace_start_all:PASS
> src/modem_trace_test.c:239:test_modem_trace_start_ip_only:PASS
> src/modem_trace_test.c:245:test_modem_trace_start_lte_ip:PASS
> src/modem_trace_test.c:254:test_modem_trace_start_with_duration_and_no_max_size:PASS
> src/modem_trace_test.c:277:test_modem_trace_stop_when_max_size_reached:PASS
> src/modem_trace_test.c:304:test_modem_trace_stop_when_received_trace_wont_fit_in_max_size_allocated:PASS
> src/modem_trace_test.c:329:test_modem_trace_stop_when_max_size_reached_before_duration:PASS
> src/modem_trace_test.c:351:test_modem_trace_stop_when_duration_reached_before_max_size:PASS
> src/modem_trace_test.c:373:test_modem_trace_forwarding_to_uart:PASS
> src/modem_trace_test.c:415:test_modem_trace_forwarding_to_rtt:PASS
> src/modem_trace_test.c:436:test_modem_trace_when_rtt_channel_allocation_fails:PASS
> src/modem_trace_test.c:455:test_modem_trace_when_transport_uart_init_fails:PASS
> src/modem_trace_test.c:474:test_modem_trace_process_when_not_initialized:PASS
> src/modem_trace_test.c:487:test_modem_trace_process_when_modem_trace_start_was_not_called:PASS
> src/modem_trace_test.c:501:test_modem_trace_start_when_nrf_modem_at_printf_fails:PASS
> src/modem_trace_test.c:512:test_modem_trace_stop_when_nrf_modem_at_printf_fails:PASS
> 
> -----------------------
> 20 Tests 0 Failures 0 Ignored 
> OK
> PROJECT EXECUTION SUCCESSFUL

To be sure I work on `docker-dtr.nordicsemi.no/sw-production/ncs-ext:2.8.2` Docker image, which is used in our Jenkins CI.

Reply:
Nice. I will add it. Side note: Should all existing unit tests be having this `integration_platforms` property then?

Reply:
What's the additional benefit or running these tests on `qemu_cortex_m3` instead of just `native_posix`?

Reply:
> Should all existing unit tests be having this `integration_platforms` property then?

@balaji-nordic Generally yes, all tests created in `testcase.yaml` and `sample.yaml` files should have this `integration_platforms` property to specify what kind of platforms should be used in CI testing. In specific or questionable situation please contact with me or @PerMac. Some time ago Maciej conducted bigger action to get `integration_platform` right in `.yaml` files: https://github.com/nrfconnect/sdk-nrf/pull/5742 and https://github.com/nrfconnect/sdk-nrf/pull/5850

Reply:
> What's the additional benefit or running these tests on `qemu_cortex_m3` instead of just `native_posix`?

@lemrey Generally QEMU is emulator which is "closer" to real hardware rather than `native_posix` simulator. So if during test there will be some problems connected to `ARM` specific architecture, this problem will be discovered faster on `qemu_cortex_m3` rather than on `native_posix`. So in my opinion testing on `qemu_cortex_m3` is closer to testing on our nRF boards rather than testing on `native_posix`. The good question is: "is this test depend so much on `ARM` specific architecture?" I don't think so. But on the other hand sometimes there are problems connected with QEMU emulator or Native-POSIX simulator. If this test will fail on one of them (and pass on second one), there are high probability that there is problem with emulator/simulator rather than with this test. So, to summarize I suggest to stay with both of this platform in `platform_allow` and `integration_platforms` :)